typing/ctype.ml

```diff
@@ -1913,7 +1913,10 @@ let occur_univar env ty =
             let td = Env.find_type p env in
             List.iter2
               (fun t v ->
-                if Variance.(mem May_pos v || mem May_neg v)
+                (* Optimization: rather than expanding the type definition
+                   for abbreviations, use the inferred presence information.
+                   Only public abbreviations may make this formula false. *)
```

Comment:
If we had bivariant annotation on abstract types, would they not make the formula false too?